src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp

```diff
@@ -133,6 +135,109 @@ namespace AppInstaller::CLI::Workflow
+            std::vector<std::string> invalidFeatures;
+
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&invalidFeatures](Dependency dependency)
+                {
+                    const auto& name = dependency.Id();
+                    WindowsFeature::WindowsFeature windowsFeature{ name };
+                    if (!windowsFeature.DoesExist())
+                    {
+                        invalidFeatures.emplace_back(name);
+                    }
+                });
+
+            if (!invalidFeatures.empty())
+            {
+                bool shouldTerminate = true;
+                auto warn = context.Reporter.Warn();
+                if (context.Args.Contains(Execution::Args::Type::IgnoreMissingDependencies))
+                {
+                    warn << Resource::String::WindowsFeatureNotFoundOverride << std::endl;
+                    shouldTerminate = false;
+                }
+                else
+                {
+                    warn << Resource::String::WindowsFeatureNotFoundOverrideRequired << std::endl;
+                }
+
+                warn << "  - " << Resource::String::WindowsFeaturesDependencies << std::endl;
+
+                for (const auto& feature : invalidFeatures)
+                {
+                    warn << "      " << feature << std::endl;
+                }
+
+                if (shouldTerminate)
+                {
+                    AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_INSTALL_MISSING_DEPENDENCY);
+                }
+            }
+
+            HRESULT hr = S_OK;
+            bool continueOnFailure = context.Args.Contains(Execution::Args::Type::Force);
+            bool rebootRequired = false;
+
+            context.Reporter.Info() << Resource::String::EnablingWindowsFeatures << std::endl;
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&hr, &continueOnFailure, &rebootRequired](Dependency dependency)
+                {
+                    if (SUCCEEDED(hr) || continueOnFailure)
+                    {
+                        auto featureName = dependency.Id();
+                        WindowsFeature::WindowsFeature windowsFeature{ featureName };
+                        if (windowsFeature.DoesExist() && !windowsFeature.IsEnabled())
+                        {
+                            hr = windowsFeature.EnableFeature();
+                            AICLI_LOG(Core, Info, << "Enabling Windows Feature " << featureName << " returned with HRESULT: " << hr);
+                            if (hr == ERROR_SUCCESS_REBOOT_REQUIRED)
+                            {
+                                rebootRequired = true;
+                            }
+                        }
+                    }
+            });
+
+            if (FAILED(hr))
+            {
+                if (continueOnFailure)
+                {
+                    context.Reporter.Warn() << Resource::String::FailedToEnableWindowsFeatureOverridden << std::endl;

```

Comment:
Other than the log, there isn't any indication of *which* feature(s) failed to enable; Users may want this information to attempt manually enabling the feature without having to dig into the log.

Reply:
Changed to report each feature that failed to enable